api/npm.ts

```diff
@@ -66,17 +67,27 @@ async function handler ({ topic, scope, pkg, tag }: PathArgs) {
   }
 }
 
-async function unpkg (topic, pkg, tag = 'latest') {
-  // const endpoint = `https://unpkg.com/${pkg}@${tag}/package.json`
-  const endpoint = `https://cdn.jsdelivr.net/npm/${pkg}@${tag}/package.json`
-  const meta = await got(endpoint).json<any>()
+async function npmMetadata (pkg: string, ver = 'latest'): Promise<any> {
+  // only works for ver === 'latest', none-scoped package
+  const endpoint = `https://registry.npmjs.org/${pkg}/${ver}`
+  return got(endpoint).json<any>()
+}
+
+async function pkgJson (pkg: string, tag = 'latest'): Promise<any> {
+  // const endpoint = `https://cdn.jsdelivr.net/npm/${pkg}@${tag}/package.json`
+  const endpoint = `https://unpkg.com/${pkg}@${tag}/package.json`
+  return got(endpoint).json<any>()
+}
+
+async function info (topic: string, pkg: string, tag = 'latest') {
+  const meta = await (tag === 'latest' && pkg[0] !== '@' ? npmMetadata(pkg) : pkgJson(pkg, tag))
 
   switch (topic) {
     case 'version': {
       return {
-        subject: `npm${tag === 'latest' ? '' : '@' + tag}`,
+        subject: tag === 'latest' ? 'npm' : `npm@${tag}`,
         status: version(meta.version),
-        color: tag === 'latest' ? versionColor(meta.version) : 'cyan'
+        color: tag === 'latest' ? versionColor(meta.version) : 'blue'
```

Comment:
hm, why always blue if not `latest`? it was cool that it's a bit different.

Reply:
I think that no matter the dist-tag it should follow the `versionColor`. But when it's on some dist-tag AND above v1 then to be cyan.

Reply:
Let's go with __no matter the dist-tag it should follow the versionColor__

Reply:
Non-latest tag will be shown on badge label:

[![](https://badgen.net/npm/v/yarn/legacy)](https://badgen.net/npm/v/yarn/legacy)

Reply:
Updated in 1999c2a

Reply:
Good.

But why just decided to switch from the cyan, in the first place? :thinking:

Reply:
`cyan` is used for some unstable/future tag (alpha, canary, rc, etc), not suitable for tag like `legacy`, so I changed it to a more general/meaningless `blue` 😋

Reply:
ooh is there still such detection in the versionColor, cool then.